UserProfile.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace GameFace.Struct
+{
+    public static class Helper
+    {
+        public static List<SprintRecords> DevideBySprints(List<UserRecords> list)
+        {
+            DateTime date = list[0].Date;
+            int quarterNumber = (date.Month - 1) / 3 + 1;
+            DateTime firstDayOfQuarter = new DateTime(date.Year, (quarterNumber - 1) * 3 + 1, 1);
+            DateTime lastDayOfQuarter = firstDayOfQuarter.AddMonths(3).AddDays(-1);
+
+            var listtemp = new List<UserRecords>();
+            var sprint = new List<SprintRecords>();
+            foreach (var record in list)
+            {
+                if (firstDayOfQuarter < record.Date && record.Date < lastDayOfQuarter)
+                {
+                    listtemp.Add(record);
+                }
+                else
+                {
+                    sprint.Add(new SprintRecords(firstDayOfQuarter, lastDayOfQuarter, listtemp));
+                    listtemp = new List<UserRecords> { record };
+                    date = record.Date;
+                    quarterNumber = (date.Month - 1) / 3 + 1;
+                    firstDayOfQuarter = new DateTime(date.Year, (quarterNumber - 1) * 3 + 1, 1);
+                    lastDayOfQuarter = firstDayOfQuarter.AddMonths(3).AddDays(-1);
+                }
+            }
+            if (listtemp.Count > 0)
+            {
+                sprint.Add(new SprintRecords(firstDayOfQuarter, lastDayOfQuarter, listtemp));
+            }
+            return sprint;
+        }
```

Comment:
While there is nothing wrong with the way you did the DevideBySprintsMethod, I'll show another way using LINQ.

```csharp
            var dividedSprints = userRecords
                .GroupBy(
                    record => new { First = record.Date.FirstDayOfQuarter(), Last = record.Date.LastDayOfQuarter() },
                    record => record)
                .Select(group =>
                {
                    var listOfUserRecords = new List<UserRecords>();
                    listOfUserRecords = group
                        .Select(record => new UserRecords(record.Task, 1, record.Value))
                        .Aggregate(listOfUserRecords, AddElement);

                    return new SprintRecords(group.Key.First, group.Key.Last, listOfUserRecords);
                })
                .ToList();
````

Unless I am mistaken, the above should be logically equivalent to your foreach. The benefit that it has over the current one is that it does not require that the elements be sorted. The downside is that it might be less clear what is happening if one is unfamiliar with some of the methods used.